.github/workflows/singularity_workflow.md

```diff
@@ -0,0 +1,57 @@
+# FastSurfer Singularity GitHub Actions Workflow
+
+This GitHub Actions workflow is designed to automate the integration testing of new code into the FastSurfer repository using Singularity containers. The workflow is triggered whenever new code is pushed to the repository.
+
+The workflow runs on a self-hosted runner labelled 'ci-gpu' to ensure security.
+
+## Jobs
+
+The workflow consists of several jobs that are executed in sequence:
+
+### Checkout
+
+This job checks out the repository using the `actions/checkout@v2` action.
+
+### Prepare Job
+
+This job sets up the necessary environments for the workflow. It depends on the successful completion of the `checkout` job. The environments set up in this job include:
+
+- Python 3.10, using the `actions/setup-python@v3` action.
+- Go, using the `actions/setup-go@v5` action with version `1.13.1`.
+- Singularity, using the `eWaterCycle/setup-singularity@v7` action with version `3.8.3`.
+
+### Build Singularity Image
+
+This job builds a Docker image and converts it to a Singularity image. It depends on the successful completion of the `prepare-job`. The Docker image is built using a Python script `Docker/build.py` with the `--device cuda --dry_run --tag fastsurfer_gpu:cuda` flags. The Docker image is then converted to a Singularity image.
```

Comment:
--dry_run will not actually build the docker image?

Reply:
`--dry_run` will not build the image, but `build.py [...] --dry_run | bash` will.

It is probably better to not use the `--dry_run` option here, which does all the steps in the background plus prints the command it would run to stdout.

Reply:
That is also, why technically the plag `--print` is a better alias to `--dry_run`.